### ISSUE TITLE
For some players, when they disconnect from server, connection is not closed for 30 seconds

### ISSUE THREAD

**Issue by cmbasnett** [DarklightGames/DarkestHour#393]
Originally reported by: **Bitbucket: [Matt_UK](http://bitbucket.org/Matt_UK), GitHub: [Matt_UK](http://github.com/Matt_UK)**

---

When player leaves the server, his body (with name) remain behind for about 30 seconds, before the server actually removes the player pawn. Player can be shot and killed during this time. After 30 seconds the player dies and is announced as leaving the server.

During the 30 seconds, if the player attempts to reconnect, the server logs a duplicate connection received and does not accept it. The player hangs on the black connecting screen and has to F10 to exit, then reconnect again. The server logs:

```
#!c++

NetComeGo: Open myLevel 09/13/15 15:17:09 192.168.1.5:61820 // new connection
Log: STEAMAUTH : Received duplicate connection for SteamID [1:12146380]
NetComeGo: Close TcpipConnection 192.168.1.5:64875 09/13/15 15:17:09 // old connection
```

When the player tries, unsuccessfully, to reconnect the 1st time, that action kills his body in game. It appears that on detecting a duplicate connection, as well as refusing the new connection, the server closes the old connection too, so properly disconnecting the player from the server. This is a Steam thing, I'm sure. But the root problem is that the player fails to disconnect from the server on exiting; all else is a symptom of that.

An old bug; at least in DH 5.1 and very likely RO as well. And for me this only happens with my main PC, all the time, but never with my old laptop. Both are using a wire connection through the same powerline adapter to the same router. I've tried with default DH.ini and User.ini files, so it's not a game config issue.

---
- Bitbucket: https://bitbucket.org/darklightgames/darkesthour/issue/394

┆Issue is synchronized with this [Trello card](https://trello.com/c/JZxXXTBY)

**Comment by handsm** [comment]
A very belated answer to Theel's question!  (Hey, I never saw it because his comment was orignally incorrectly formatted & so most of the text disappeared off the edge of the screen without wrapping.) :)

This happens when purposely disconnecting from a server, not just when a net connection is lost. I tested lowering the `ConnectionTimeout` value from the default 30 seconds and it works. I tried it on 1 second, which meant my pawn died and I properly disconnected from the server almost immediately, allowing me to rejoin straight away. But at that very low setting I did begin to suffer from unwanted disconnects after idling. So a value of perhaps 10 seconds would probably work ok.
